src/layers/WellboreBaseComponentLayer.ts

```diff
@@ -157,47 +140,45 @@ export abstract class WellboreBaseComponentLayer extends PixiLayer {
     this.ctx.stage.addChild(rope);
   }
 
-  drawRope(path: Point[], texture: Texture): void {
+  drawRope(path: Point[], texture: Texture, tint?: number): void {
     if (path.length === 0) {
       return null;
     }
 
     const rope: SimpleRope = new SimpleRope(texture, path, 1);
+
+    if (tint) {
```

Comment:
A small thing, but here I would have used a ternary expression to make it clear what rope.tint will be.
`rope.tint = tint || null` eller `rope.tint = tint || rope.tint` or whatever rope.tint will be in case tint is falsy.